File: frameworks/Java/fair/java-vertx-pg-client/src/main/java/io/techempower/benchmark/vertx/util/JteUtils.java
package io.techempower.benchmark.vertx.util;

import gg.jte.generated.precompiled.JtefortunesGenerated;
import gg.jte.html.HtmlTemplateOutput;
import gg.jte.html.OwaspHtmlTemplateOutput;
import gg.jte.output.Utf8ByteOutput;
import io.techempower.benchmark.vertx.model.Fortune;

import java.util.List;

public final class JteUtils {

    private JteUtils() {
    }

    public static byte[] serializeStandard(List<Fortune> fortunes) {
        Utf8ByteOutput out = new Utf8ByteOutput(256);
        HtmlTemplateOutput template = new OwaspHtmlTemplateOutput(out);
        JtefortunesGenerated.render(template, null, fortunes);
        return out.toByteArray();
    }
}

File: frameworks/Java/fair/kotlin-ktor-cio-jdbc/src/main/kotlin/io/techempower/benchmark/ktor/util/JteUtils.kt
package io.techempower.benchmark.ktor.util

import gg.jte.generated.precompiled.JtefortunesGenerated
import gg.jte.html.OwaspHtmlTemplateOutput
import gg.jte.output.Utf8ByteOutput
import io.techempower.benchmark.ktor.model.Fortune

object JteUtils {
    fun serializeStandard(fortunes: List<Fortune>): ByteArray {
        val output = Utf8ByteOutput(256)
        JtefortunesGenerated.render(OwaspHtmlTemplateOutput(output), null, fortunes)
        return output.toByteArray()
    }
}
